### PR TITLE
Test/rendering html with nunjucks for unit tests

### PR DESCRIPTION
**Pull request by thomkainos**
### Describe your changes

This pr creates unit tests for the index and view job role views with nunjucks rendering. The unit tests integrate [cheerios](https://cheerio.js.org/docs/intro) library to traverse rendered html files. 

#### General
- [x] I have performed a self-review of my code.
- [x] Code changes are considered complete for the given story (i.e., the comments from code review are addressed).
- [x] The commit title follows conventional commit standards.
- [x] I have put the PR link into the planner.
- [x] The PR has a clear and descriptive title and a detailed description of the changes.
- [x] Code builds cleanly without any errors or warnings.
 
#### Code Quality
- [x] I have run linting and there were no errors.
 
#### Testing
- [x] New functionality is tested.
- [x] All unit and integration tests passed.
- [x] I have run accessibility tests and there were no errors.
 
#### Deployment
- [x] Documentation has been updated where needed.
 
#### Security
- [x] No sensitive data is hard-coded or exposed.

### CODE REVIEW

File: test/unit/views/indexTest.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    });
+
+    it('should use the correct style sheet', () => {
+        const context = {};
+        const renderedHtml = nunjucks.render(templateFile, context);
+
+        const expectedStyleSheet = 'style.css';
+
+        expect(renderedHtml).to.include(expectedStyleSheet);
+    });
```
Comment (swalker125) [inline test/unit/views/indexTest.ts]: is this test useful? Is it showing we're using style.css and not some other css file? What about when we need another css file?

File: test/unit/views/listJobRolesTest.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                { roleName: 'Software Engineer', location: 'Toronto', capability: 'Development', band: 'B2', closingDate: '2024-08-01', roleStatus: 1 },
+                { roleName: 'Product Owner', location: 'Toronto', capability: 'Business', band: 'B3', closingDate: '2024-08-01', roleStatus: 1 },
+                { roleName: 'Manager', location: 'Toronto', capability: 'Management', band: 'B1', closingDate: '2024-08-01', roleStatus: 1 },
+            ]
+        };
+
+        const renderedHtml = nunjucks.render(templateFile, context);
+        const $ = cheerio.load(renderedHtml);
+
+        const actualNumDataRows = $('table').find('tr').length - 1; // exclude header row
```
Comment (swalker125) [inline test/unit/views/listJobRolesTest.ts]: I don't like this -1. I think we should use some cheerio filtering logic: https://cheerio.js.org/docs/basics/traversing#filtering-elements

**Review by erinmcj — DISMISSED** [review]

**Review by swalker125 — COMMENTED** [review]
minor comments

File: test/unit/views/indexTest.ts (hunk 1 above)
Comment (thomkainos) [reply test/unit/views/indexTest.ts]: No it doesn't we could update it to include the link tag with the ref attribute for stylesheet. We can further extend it check against an array of style sheets?

File: test/unit/views/listJobRolesTest.ts (hunk 2 above)
Comment (thomkainos) [reply test/unit/views/listJobRolesTest.ts]: Ok, did some refactoring

**Review by swalker125 — APPROVED** [review]
LGTM

**Review by erinmcj — APPROVED** [review]

**Pull request merged**